hydrosdk/deployment_configuration.py

```diff
@@ -0,0 +1,473 @@
+        else:
+            self.container_spec.resources = new_resource_requirements
+        return self
+
+    def with_toleration(self,
+                        effect: str = None,
+                        key: str = None,
+                        operator: str = "Equal",
+                        toleration_seconds: int = None,
+                        value: str = None) -> 'DeploymentConfigurationBuilder':
+        """
+        Specify pod's toleration. The pod this Toleration is attached to tolerates any taint that matches the triple <key,value,effect>
+         using the matching operator <operator>.
+
+        :param effect: Effect indicates the taint effect to match.
+            Empty means match all taint effects. When specified, allowed values are NoSchedule, PreferNoSchedule and NoExecute.
+        :param key: Key is the taint key that the toleration applies to. Empty means match all taint keys.
+         If the key is empty, operator must be Exists; this combination means to match all values and all keys.
+        :param operator: Operator represents a key's relationship to the value. Valid operators are Exists and Equal. Defaults to Equal.
+         Exists is equivalent to wildcard for value, so that a pod can tolerate all taints of a particular category.
+        :param toleration_seconds: TolerationSeconds represents the period of time the toleration (which must be of effect NoExecute,
+         otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint forever (do not evict).
+         Zero and negative values will be treated as 0 (evict immediately) by the system.
+        :param value: Value is the taint value the toleration matches to. If the operator is Exists, the value should be empty,
+         otherwise just a regular string.
+        """
+        new_toleration = Toleration(effect=effect, key=key, operator=operator, toleration_seconds=toleration_seconds, value=value)
+        if self.pod_spec is None:
+            self.pod_spec = PodSpec(tolerations=[new_toleration])
+        else:
+            self.pod_spec.tolerations.append(new_toleration)
+        return self
+
+    def with_replicas(self, replica_count=1) -> 'DeploymentConfigurationBuilder':
+        """
+        Specify number of desired pods for a deployment
+
+        :param replica_count: Number of desired pods. This is a pointer to distinguish between explicit zero and not specified.
+         Defaults to 1.
+        :return:
+        """
+        if self.deployment_spec is None:
+            self.deployment_spec = DeploymentSpec(replica_count=replica_count)
+        elif self.deployment_spec.replica_count is not None:
+            raise ValueError("Replica Count is already set")
+        else:
+            self.deployment_spec.replica_count = replica_count
+        return self
+
+    def build(self) -> DeploymentConfiguration:
```

Comment:
ApplicationBulider immediately sends POST request to create a new application. We should either do that here as well, or remove POST behaviour from ApplicationBuilder.

Reply:
Can you elaborate? `DeploymentConfigurationBuilder` sends a `POST` request after `build()` is called. 

https://github.com/Hydrospheredata/hydro-serving-sdk/blob/bf2b769f406acf8e7364434e715906c0e72c4a4f/hydrosdk/deployment_configuration.py#L471